src/python/pants/build_graph/target.py

```diff
@@ -868,6 +868,12 @@ def create_sources_field(self, sources, sources_rel_path, key_arg=None):
       # This is so that tests don't need to set up the subsystem when creating targets that
       # legitimately do not require sources.
       if (key_arg is None or key_arg == 'sources') and self.supports_default_sources():
+        logger.warning(
```

Comment:
Should this use a (long) deprecation instead?

Reply:
Yeah... I originally had a deprecation, but changed it to a warning because I couldn't point people at actual problems or alternatives, and really just want them to get in touch because a supposedly unreachable situation has happened...

Happy to go either way... Thoughts?

Reply:
I think that eventually we'll want to remove this warning, and a deprecation is a good way to trigger that removal. If there is some other refactoring that will cause the case that the warning is triggering to become impossible, could link there and skip the deprecation I guess.